wherehows-web/app/components/datasets/health/score-table.ts

```diff
@@ -0,0 +1,182 @@
+  class: string;
+  dropdownOptions: IDropDownOption<string>[] | undefined;
+  initialDropdown: IDropDownOption<string>;
+  onDropdownSelect: (type: string, selection: IDropDownOption<string>) => void;
+}
+
+/**
+ * Defining a loose interface for the dropdowns object calculated for category and severity based on
+ * the received table data
+ */
+interface IHealthTableDropdowns {
+  [key: string]: Array<IDropDownOption<string>>;
+}
+
+export default class DatasetsHealthScoreTable extends Component {
+  /**
+   * Sets the class names binded to the html element generated by this component
+   * @type {Array<string>}
+   */
+  classNames = ['nacho-table', 'dataset-health__score-table'];
+
+  /**
+   * Sets the tag to be rendered for the html element generated by this component
+   * @type {string}
+   */
+  tagName = 'table';
+
+  /**
+   * Expected headers for the detailed table. It contains the wording for each header as well as
+   * additional properties
+   * @type {Array<IHealthTableHeader>}
+   */
+  headers = computed('dropdownOptions', function(this: DatasetsHealthScoreTable): Array<IHealthTableHeader> {
+    const dropdownOptions = get(this, 'dropdownOptions');
+    const onDropdownSelect = get(this, 'onDropdownSelect');
+
+    return ['Category', 'Description', 'Score', 'Severity'].map(header => ({
+      label: header,
+      class: `dataset-health__score-table__${header.toLowerCase()}`,
+      dropdownOptions: dropdownOptions[header.toLowerCase()],
+      initialDropdown: { label: header, value: '' },
+      onDropdownSelect: onDropdownSelect.bind(this, header)
+    }));
+  });
+
+  /**
+   * Uses the passed in table data to calculate a dropdown list of options available for filtering by category
+   * and severity for the table's rows.
+   * @type {ComputedProperty<IHealthTableDropdowns>}
+   */
+  dropdownOptions: ComputedProperty<IHealthTableDropdowns> = computed('tableData', function(
+    this: DatasetsHealthScoreTable
+  ) {
+    const tableData = get(this, 'tableData');
+    const includedOptions = new Set();
+
+    return tableData.reduce(
+      (output, row) => {
+        const { category, severity } = row;
+
+        !includedOptions.has(category) && output.category.push({ label: category, value: category });
+        severity && !includedOptions.has(severity) && output.severity.push({ label: severity, value: severity });
+        // Ensures we only add each option once, assumes severity will never === category
+        includedOptions.add(category);
+        includedOptions.add(severity);
+
+        return output;
+      },
+      {
+        category: <IDropDownOption<string>[]>[{ label: 'Category', value: '' }],
+        severity: <IDropDownOption<string>[]>[{ label: 'Severity', value: '' }]
+      }
+    );
+  });
+
+  /**
+   * Passed in table data, mostly raw detailed information about the compliance score details and the
+   * breakdown of such.
+   * @type {Array<IHealthScore>}
+   */
+  tableData: Array<IHealthScore>;
```

Comment:
let's keep property declarations above definitions, i.e. `tableData`, `currentSeverityFilter `, `currentCategoryFilter ` above ComputedProperty definitions, it's visually clearer when glancing at the structure of a class to see the declarations & class fields first, before 'beefier' properties and methods